pyomo/contrib/doe/documentation/docs/source/index.rst

```diff
@@ -0,0 +1,453 @@
+      -   :math:`\text{det}({\mathbf{M}})`
+      - Volume of the confidence ellipse
+    * - E-optimality
+      -   :math:`\text{min eig}({\mathbf{M}})`
+      - Size of the longest axis of the confidence ellipse
+    * - Modified E-optimality
+      -   :math:`\text{cond}({\mathbf{M}})`
+      - Ratio of the longest axis to the shortest axis of the confidence ellipse
+
+In order to solve problems of the above, Pyomo.DOE implements the 2-stage stochastic program. Please see Wang and Dowling (2022) for details.
+
+Pyomo.DOE Required Inputs
+--------------------------------
+The required inputs to the Pyomo.DOE solver are the following:
+
+* A function that creates the process model
+* Dictionary of parameters and their nominal value
+* Dictionary of measurements and their measurement time points
+* Dictionary of design variables and their control time points
+* Prior FIM
+* Subordinate local and global NLP optimization solver
+
+Below is a list of arguments that Pyomo.DOE expects the user to provide.
+
+param_init : ``dictionary``
+    A ``dictionary`` of parameter names and values. If they are an indexed variable, put the variable name and index in a nested ``Dictionary``.
+
+design_variable_timepoints : ``dictionary``
+    A ``dictionary`` of design variable names and its control time points. If this design var is independent of time (constant), set the time to [0]
+
+measurement_object : ``object``
+    A ``object`` of the measurements, provided by the measurement class.
+
+
+create_model : ``function``
+    A ``function`` representing a deterministic process model.
+
+
+prior_FIM : ``list``
+    A ``list`` defining the Fisher information matrix (FIM) for prior experiments, default is a zero matrix.
+
+Pyomo.DOE Solver Interface
+---------------------------
+
+.. figure:: uml.png
+   :scale: 25 %
+   :alt: map to buried treasure
+
+
+.. autoclass:: fim_doe.DesignOfExperiments
+    :members:  __init__, optimize_doe, compute_FIM, run_grid_search
+
+
+.. Note::
+    Optimize_doe() includes the following steps:
+        #.  Build two-stage stochastic programming optimization model where scenarios correspond to finite difference approximations for the Jacobian of the response variables with respect to calibrated model parameters
+        #.  Fix the experiment design decisions and solve a square (i.e., zero degrees of freedom) instance of the two-stage DOE problem. This step is for initialization.
+        #.  Unfix the experiment design decisions and solve the two-stage DOE problem.
+
+.. autoclass:: fim_doe.Measurements
+    :members: __init__, check_subset
+
+.. autoclass:: fim_doe.Scenario_generator
+    :special-members: __init__
+
+.. autoclass:: fim_doe.FIM_result
+    :special-members: __init__, calculate_FIM
+
+.. autoclass:: fim_doe.Grid_Search_Result
+    :special-members: __init__
+
+
+
+Pyomo.DOE Usage Example
+------------------------
+
+We will use an example to illustrate the usage of Pyomo.DOE to a reaction kinetics example.
```

Comment:
We illustrate the usage of Pyomo.DOE using a reaction kinetics example (Wang and Dowling, 2022).